test/e2e/kubectl.go

```diff
@@ -97,11 +97,11 @@ var _ = Describe("kubectl", func() {
 	Describe("guestbook", func() {
 		var guestbookPath = filepath.Join(testContext.RepoRoot, "examples/guestbook")
 
-		if testContext.Provider != "gce" && testContext.Provider != "gke" {
-			By(fmt.Sprintf("Skipping guestbook, uses createExternalLoadBalancer, a (gce|gke) feature"))
-		}
-
 		It("should create and stop a working application", func() {
+			if testContext.Provider != "gce" && testContext.Provider != "gke" {
+				By(fmt.Sprintf("Skipping guestbook, uses createExternalLoadBalancer, a (gce|gke) feature"))
```

Comment:
also, does this actually skip anything? It seems like all this does is print out the By line, but doesn't actually have any logic to really skip (unless that's hidden somewhere else?)